examples/minnietwitter/src/main/java/amino/run/appexamples/minnietwitter/MinnieTwitterMain.java

```diff
@@ -142,46 +142,159 @@ private static void ExecuteSingleUserDemo(InetSocketAddress hostAddr, InetSocket
     }
   }
 
-  private static void ExecuteFullDemo(InetSocketAddress hostAddr, InetSocketAddress omsAddr) {
+  /**
+   * Create an instance of the TwitterManager Microservice
+   * @param oms
+   * @return
+   * @throws RemoteException
+   * @throws MicroServiceCreationException
+   * @throws MicroServiceNotFoundException
+   */
+  private static TwitterManager createTwitterManager(Registry oms, String microServiceName) throws RemoteException, MicroServiceCreationException, MicroServiceNotFoundException, MicroServiceNameModificationException {
+    /* Get Twitter and User Manager */
+    MicroServiceSpec spec =
+            MicroServiceSpec.newBuilder()
+                    .setLang(Language.java)
+                    .setJavaClassName("amino.run.appexamples.minnietwitter.TwitterManager")
+                    .create();
+
+    MicroServiceID microServiceId = oms.create(spec.toString());
+    /* To set a name to microservice. It is required to set the name if the object has to be shared */
+    oms.setName(microServiceId, microServiceName);
+    return (TwitterManager) oms.acquireStub(microServiceId);
 
-    java.rmi.registry.Registry registry;
-    List<Timeline> timelines = new ArrayList<Timeline>();
+  }
 
-    try {
-      registry = LocateRegistry.getRegistry(omsAddr.getHostString(), omsAddr.getPort());
-      Registry server = (Registry) registry.lookup("io.amino.run.oms");
+  /**
+   * Check peer tweets on an existing twitter microservice.
+   * This is based on TwitterActivityOne in the original Sapphire examples.
+   */
+  private static void checkPeerTweets(TwitterManager tm) {
+    UserManager usrMan = tm.getUserManager();
+    TagManager tagMan = tm.getTagManager();
+    for(int i=0; i<USER_COUNT; i++) {
+      String userName = "user" + Integer.toString(i);
+      User user = usrMan.getUser(userName);
+      Timeline timeline = user.getTimeline();
+      List<Tweet> peerTweets = timeline.getTweets(0, EVENTS_PER_USER);
+      System.out.println("User " + userName + " has " + peerTweets.size() + " tweets");
+
+      for (Tweet t : peerTweets) {
+        System.out.println("Tweet: '" + t.getText() +"'; number of retweets: " + t.getRetweetes() + "; number of favorites: " + t.getFavorites());
+      }
+    }
+  }
 
-      KernelServer nodeServer = new KernelServerImpl(hostAddr, omsAddr);
+  /**
+   * Check that following works on an existing twitter microservice.
+   * This is based on TwitterActivityTwo in the original Sapphire examples.
+   */
+  private static void checkFollowing(TwitterManager tm) {
+    UserManager usrMan = tm.getUserManager();
```

Comment:
checkPeerTweets function also gets the manager. I am wondering if it will be better to set it as a class variable rather than getting it at every function.

Reply:
I did this intentionally, to simulate two different clients.